engines/zypper_auth/spec/requests/strict_authentication/authentication_controller_spec.rb

```diff
@@ -204,6 +206,41 @@
 
         it { is_expected.to have_http_status(200) }
       end
+
+      context 'system is hybrid' do
+        include_context 'auth header', :system_hybrid, :login, :password
+        let(:system_hybrid) { FactoryBot.create(:system, :hybrid, :with_activated_product) }
+        let(:requested_uri) { '/repo' + system_hybrid.repositories.first[:local_path] + '/repodata/repomd.xml' }
+        let(:headers) { auth_header.merge({ 'X-Original-URI': requested_uri, 'X-Instance-Data': 'test' }) }
+        let(:scc_response) do
+          {
+            is_active: false,
+            message: 'You shall not have access to those repos !'
+          }
+        end
+
+        context 'regcode check fails' do
+          before do
+            Rails.cache.clear
+            expect_any_instance_of(InstanceVerification::Providers::Example).to receive(:instance_valid?).and_return(true)
+            allow(InstanceVerification).to receive(:update_cache)
+            allow(SccProxy).to receive(:scc_check_subscription_expiration).and_return(scc_response)
+            expect(SccProxy).to receive(:scc_check_subscription_expiration)
+            allow(File).to receive(:directory?)
```

Comment:
Which part of the code does create a file on disk when you check a regcodes validity

Reply:
the method that creates the cache file on disk [update_cache](https://github.com/SUSE/rmt/blob/master/engines/instance_verification/lib/instance_verification/engine.rb#L4) called [here](https://github.com/SUSE/rmt/blob/master/engines/zypper_auth/lib/zypper_auth/engine.rb#L35) for example